src/server/scripts/EasternKingdoms/BlackrockMountain/BlackrockSpire/boss_overlord_wyrmthalak.cpp

```diff
@@ -77,7 +77,12 @@ class boss_overlord_wyrmthalak : public CreatureScript
 
         void JustDied(Unit* /*killer*/) override
         {
-            _JustDied();
+            events.Reset();
+            if (instance)
+            {
+                instance->SetBossState(DATA_OVERLORD_WYRMTHALAK, DONE);
```

Comment:
this is the only line actually required

Reply:
I tried to keep the change with "minimal impact". I believe only the SaveToDB could have any use, the boss state being unused as far as I'm aware...

Reply:
I understand, but SetBossState() already saves to database: https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Instances/InstanceScript.cpp#L331

And events.Reset() has no use because the creature is dead and events reset when they respawn on Reset()

Furthermore, boss scripts never run outside instances, so check instance for nullptr is also trivial

Reply:
> Furthermore, boss scripts never run outside instances, so checking instance for nullptr is also trivial.

Hmm I doubt that part (WBs in open world come to my mind), but else we can simplify bossAI. That's where I got the code from. (since I removed the call to _JustDied() )
![image](https://user-images.githubusercontent.com/2592673/140657374-19c1297a-844d-4b49-a564-68164c2f9617.png)

Reply:
See this  GetBlackrockSpireAI macro?

https://github.com/azerothcore/azerothcore-wotlk/pull/9036/files#diff-9a42647dacaa111473022e06147c4c23dcddcd80d42a63c0a86bc896ab559ee7R54

It ensures his script is only executed inside BlackrockSpire. I was wrong, he's fine, no updating required, we can get rid of the instance nullptr check.

Reply:
Ah, but then we *can* remove it if we know we're in an instance, or do I get that wrong?

Reply:
> Ah, but then we _can_ remove it if we know we're in an instance, or do I get that wrong?

I think maybe we got confused here. We can't remove it from BossAI::_JustDied() because as you said they can be world bosses.

But we don't need to check for instance script validity when bosses are registered using that kind of function (I believe every instance has, at least should have, one of those functions to check if the creature is in the right map).

So yes, we can remove it from his script

Reply:
Ok, then I remove it here while changing up a few other things in there.